arangod/GeneralServer/AuthenticationFeature.cpp

```diff
@@ -166,22 +172,27 @@ void AuthenticationFeature::prepare() {
     _userManager.reset(new auth::UserManager(server()));
 #endif
   } else {
-    LOG_TOPIC("713c0", DEBUG, Logger::AUTHENTICATION) << "Not creating user manager";
+    LOG_TOPIC("713c0", DEBUG, Logger::AUTHENTICATION)
+        << "Not creating user manager";
   }
 
   TRI_ASSERT(_authCache == nullptr);
   _authCache.reset(new auth::TokenCache(_userManager.get(), _authenticationTimeout));
 
-  std::string jwtSecret = _jwtSecretProgramOption;
-  if (jwtSecret.empty()) {
+  if (_jwtSecretProgramOption.empty()) {
     LOG_TOPIC("43396", INFO, Logger::AUTHENTICATION)
         << "Jwt secret not specified, generating...";
     uint16_t m = 254;
     for (size_t i = 0; i < _maxSecretLength; i++) {
-      jwtSecret += (1 + RandomGenerator::interval(m));
+      _jwtSecretProgramOption += static_cast<char>(1 + RandomGenerator::interval(m));
```

Comment:
Should we exclude the space character (0x20), the tab, CR and LF characters here?
In case a generated secret contains these characters at the beginning or end, they will be trimmed away next time the secret is read?

Reply:
this secret is not written to disk I think